test/rmath.jl

```diff
@@ -17,7 +17,18 @@ end
 get_statsfun(fname) = eval(Symbol(fname))
 get_rmathfun(fname) = eval(Meta.parse(string("RFunctions.", fname)))
 
-function rmathcomp(basename, params, X::AbstractArray, rtol=100eps(float(one(eltype(X)))))
+function rmathcomp(basename, params, X::AbstractArray, rtol=nothing)
+    # compute desired tolerance:
+    # has to take into account `params` as well since otherwise e.g. `X::Array{<:Rational}`
+    # always uses a tolerance based on `eps(one(Float64))` even when parameters are of type
+    # Float32
+    rtol = if rtol === nothing
+        PT = mapreduce(typeof, promote_type, params)
```

Comment:
Couldn't you use one of the promotion function in Base here?

Reply:
Is there one that promotes `(1, 1//2)` or `(1f0, 2.0)` to `Rational{Int}` and `Float64`?

Reply:
```julia
julia> Base.promote_typeof(1, 1//2)
Rational{Int64}

julia> Base.promote_typeof(1f0, 2.0)
Float64
```

```suggestion
        PT = Base.promote_typeof(params...)
```

Reply:
I had in mind that splatting in such a case is much slower (generally). And indeed, e.g.,
```julia
julia> using BenchmarkTools

julia> f(x) = Base.promote_typeof(x...)
f (generic function with 1 method)

julia> g(x) = mapreduce(typeof, promote_type, x)
g (generic function with 1 method)

julia> x = ntuple(_ -> (c = rand(); c < 0.2 ? rand(Float64) : (c < 0.7 ? rand(Int) : rand(Int) // rand(Int))), 100);

julia> @btime f($x);
  94.985 μs (101 allocations: 3.14 KiB)

julia> @btime g($x);
  18.967 μs (101 allocations: 3.14 KiB)
```
However, I guess it does not matter here since (a) it's only for tests and (b) the number of parameters, i.e., length of the tuples, is very small (I think <= 3 at most). So I'll use your suggestion to improve readability of the code :slightly_smiling_face: